src/macrofier/engine.mjs

```diff
@@ -1140,46 +1214,47 @@ function insertExampleMacros(template, examples, method, json, templates) {
   return template.replace(/\$\{method\.examples\}/g, content)
 }
 
-function generateResult(result, json, templates) {
-  const type = types.getSchemaType(result, json, { destination: state.destination, section: state.section  })
+function generateResult(result, json, templates, { name = '' } = {}) {
+
+  const type = types.getSchemaType(result, json, { name: name, destination: state.destination, section: state.section  })
 
   if (result.type === 'object' && result.properties) {
     let content = getTemplate('/types/object', templates).split('\n')
 
     for (var i=0; i<content.length; i++) {
       if (content[i].indexOf("${property}") >= 0) {
-        content[i] = Object.entries(result.properties).map(([title, property]) => insertSchemaMacros(content[i], title, property, json)).join('\n')
+        content[i] = Object.entries(result.properties).map(([title, property]) => insertSchemaMacros(content[i], title || name, property, json)).join('\n')
```

Comment:
The `name` variable is the top-level name passed in via the method call.
The `title` variable is the individual title of each property the schema contains.

The `result.property` object will ALWAYS have name/value pairs, so the `title` variable above will always be populated. I think you can drop the `|| name`

Reply:
will update